app/models/playlist_item.rb

```diff
@@ -26,4 +26,19 @@ class PlaylistItem < ActiveRecord::Base
   before_save do
     clip.save
   end
+
+  def replicate
```

Comment:
How about making this replicate! and changing all of the .save calls to .save! so exceptions are raised and bubble up instead of being silently skipped over.